src/modules/http-exchange.ts

```diff
@@ -23,6 +23,13 @@ const createAbortController = () => {
   return new AbortController();
 };
 
+// default headers to be used in functional fetch options in 'client.ts'
+// freezing the object so that a consumer of the library does not
+// use a mutable change on headers causing unexpected side effects on other http calls.
+export const defaultHeaders: {} = Object.freeze({
```

Comment:
I'm not sure whether this sacrifices some compat or pulls in another polyfill. Either might be bad :/ Gotta check first. But I think it'd be save to just call `fetchOptions` with a fresh default object every time as that's just as save